sys/include/net/nanocoap_sock.h

```diff
@@ -9,13 +9,14 @@
 /**
```

Comment:
would it make sense to split this into `nanocoap/client.h` and `nanocoap/server.h`?

Reply:
I agree with this, but I don't mind leaving this to a follow up PR. @kb2ma What do you think?

Reply:
> would it make sense to split this into `nanocoap/client.h` and `nanocoap/server.h`?

I like the 'name substitution' aspect. In other words, get rid of "sock" from the name, which is descriptive but overlaps with the name of the messaging stack. The use of 'client' and 'server' imply endpoint messaging without really conflicting with gcoap. It's possible that `nanocoap.h` might split someday for other reasons, but I think we can manage the distinction from the client/server headers. The module documentation title can change from 'Nanocoap Sock' to 'Nanocoap Client/Server' or something like that.

In terms of implementation files, remember that the client/server split can be murky. A client might send a NON request but receive a CON response and have to manage retries. Neither gcoap or nanocoap supports that now, but it's out there.

Reply:
> I agree with this, but I don't mind leaving this to a follow up PR. @kb2ma What do you think?

That works for me. Let's make the migration out of nanocoap.(h|c) itself as one step, and then the group that works more closely with it can split it up however works best.